File: src/main/java/ru/pearx/pmdumper/dumpers/DumperBlocks.java
package ru.pearx.pmdumper.dumpers;

import net.minecraft.block.Block;
import net.minecraft.client.Minecraft;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.common.registry.ForgeRegistries;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import ru.pearx.pmdumper.dumpers.base.IDumper;
import ru.pearx.pmdumper.dumpers.base.PMDData;

import java.lang.reflect.Array;
import java.util.*;

/**
 * Created by mrAppleXZ on 13.06.17 19:53.
 */
@SideOnly(Side.CLIENT)
public class DumperBlocks implements IDumper
{
    @Override
    public String getName()
    {
        return "blocks";
    }

    @Override
    public PMDData getData()
    {
        Map<String, Integer> counts = new HashMap<>();
        List<List<String>> lst = new ArrayList<>();
        lst.add(Arrays.asList("ID", "Properties", "Class Name"));
        for(Map.Entry<ResourceLocation, Block> entr : ForgeRegistries.BLOCKS.getEntries())
        {
            ResourceLocation loc = entr.getKey();
            Block block = entr.getValue();
            lst.add(Arrays.asList(loc.toString(),
                    block.getBlockState().getProperties().toString(),
                    block.getClass().getName()));
            PMDData.plusCounts(counts, loc.getResourceDomain(), 1);
        }
        return new PMDData(lst, counts);
    }
}

File: src/main/kotlin/ru/pearx/pmdumper/dumper/standard/DumperBlocks.kt
@file:JvmMultifileClass
@file:JvmName("StandardDumpers")
package ru.pearx.pmdumper.dumper.standard

import net.minecraft.util.ResourceLocation
import net.minecraftforge.fml.common.registry.ForgeRegistries
import ru.pearx.pmdumper.ID
import ru.pearx.pmdumper.dumper.dumper

val DumperBlocks = dumper {
    registryName = ResourceLocation(ID, "blocks")
    header = listOf("ID", "Class Name", "BlockState Properties", "BlockState Class Name")
    amounts {
        for (block in ForgeRegistries.BLOCKS)
            this += block.registryName
    }
    iterator {
        for (block in ForgeRegistries.BLOCKS) {
            with(ArrayList<String>(header.size)) {
                with(block) {
                    add(registryName.toString())
                    add(this::class.java.name)
                    add(blockState.properties.toString())
                    add(blockState::class.java.name)
                }
                yield(this)
            }
        }
    }
}